.github/workflows/schema.yaml

```diff
@@ -0,0 +1,26 @@
+name: schema
+on: [pull_request]
+
+jobs:
+  generate-schema:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout rockcraft
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install requirements
+        run: pip install -r requirements.txt
+
+      - name: Run generator script
+        run: python .github/scripts/schema.py > generated_schema.json
+
+      - name: Check with stored schema
+        run: diff generated_schema.json schemas/rockcraft_schema.json > /dev/null
```

Comment:
Why `/dev/null`? It would be useful to know the mismatch when an error occurs...